tests/test_cli.py

```diff
@@ -131,5 +132,26 @@ def test_cli_reloader_incomplete_app_parameter(
     ) in captured.err
 
 
+def test_app_dir(tmp_path: Path, caplog: pytest.LogCaptureFixture) -> None:
+    app_dir = tmp_path / "dir" / "app_dir"
+    app_file = app_dir / "main.py"
+    app_dir.mkdir(parents=True)
+    app_file.touch()
+    app_file.write_text(
+        dedent(
+            """
+            async def app(scope, receive, send):
+                ...
+            """
+        )
+    )
+    runner = CliRunner()
+    with mock.patch.object(main, "run") as mock_run:
+        result = runner.invoke(cli, ["main:app", "--app-dir", f"{str(app_dir)}"])
+
+    assert result.exit_code == 0
+    mock_run.assert_called_once()
```

Comment:
I don't think this is a valid test for us. I can replace the value for `--app-dir` with anything and it will still pass the assertions. See below: 

```python
def test_app_dir(tmp_path: Path, caplog: pytest.LogCaptureFixture) -> None:
    runner = CliRunner()
    with mock.patch.object(main, "run") as mock_run:
        result = runner.invoke(cli, ["main:app", "--app-dir", "potato"])

    assert result.exit_code == 0
    mock_run.assert_called_once()
```